app/models/dialog_field_sorted_item.rb

```diff
@@ -101,19 +101,28 @@ def use_first_value_as_default
     self.default_value = sort_data(@raw_values).first.try(:first)
   end
 
-  def default_value_included_in_raw_values?
-    @raw_values.collect { |value_pair| value_pair[0].send(value_modifier) }.include?(default_value.send(value_modifier))
+  def default_value_included?(values_list)
+    values_list.collect { |value_pair| value_pair[0].send(value_modifier) }.include?(default_value.send(value_modifier))
   end
 
   def static_raw_values
-    first_values = required? ? [[nil, "<Choose>"]] : initial_values
-    first_values + self[:values].to_miq_a.reject { |value| value[0].nil? }
+    self[:values].to_miq_a.reject { |value| value[0].nil? }.unshift(nil_option).reject(&:empty?)
   end
 
   def initial_values
     [[nil, "<None>"]]
   end
 
+  def nil_option
+    if !required?
+      initial_values.flatten
+    elsif default_value.blank? || !default_value_included?(self[:values])
+      [nil, "<Choose>"]
```

Comment:
@eclarizio - For consistency - possibly move this out into a method like `initial_values` - Or bring both back into `nil_option`